### DISCUSSION TITLE
Redisson storing value in hash form

### DISCUSSION THREAD

**Discussion by gsinhazscaler** [redisson/redisson#5613] [General]
When I am trying to use Redisson library it always store data in hash form. Is there anyway I can avoid it if I like

        Config config = new Config();
        config.useSingleServer()
                .setAddress("redis://127.0.0.1:6379");
        RedissonClient redisson = Redisson.create(config);
        System.out.println("Config created"+redisson);
        RBucket<String> r=redisson.getBucket("test");
        r.set("value", 1000000, TimeUnit.SECONDS);
Here is how I see the data in Redis

127.0.0.1:6379> Get test
"\x03valu\xe5"
How can I view the data properly from the redis cli? Or how can I avoid storing the value in hash form completely

**Comment by mrniko** [comment]
You need to use StringCodec. Default one is a binary Kryo codec.

**Reply by gsinhazscaler** [reply]
@mrniko : I did apply but I have a follow up question

**Comment by gsinhazscaler** [comment]
![image (2)](https://github.com/redisson/redisson/assets/90656654/aa5115b1-ca66-4598-8b43-0a75460dddec)

**Comment by gsinhazscaler** [comment]
RMapCache<String, T> cache = client.getMapCache(cache,StringCodec.INSTANCE);
cache.fastPut(key,val, sec, TimeUnit.SECONDS);

I still see a bit of   "\x00\x00" even after doing this change

**Reply by mrniko** [reply]
MapCache object stores some extra data per value to track idle time.
